library/Fission/Web/Middleware/AuthBody.hs

```diff
@@ -0,0 +1,39 @@
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
+
+module Fission.Web.Middleware.AuthBody (foo) where
+
+import Fission.Prelude
+
+import Network.Wai (Middleware, strictRequestBody)
+import Network.Wai.Internal
+
+import qualified Data.ByteString.Lazy       as Lazy
+
+import qualified Data.Vault.Lazy as Vault
+
+-- type Middleware = Application -> Application
+-- type Application =
+--    Request -> (Response -> IO ResponseReceived) -> IO ResponseReceived
+
+-- how to generate a unique vault key outside this function:
+-- `vaultKey <- newKey`
+
+foo :: Vault.Key Text -> Middleware
+foo vaultKey app req sendResponse = do
+  lazyBody <- strictRequestBody req
```

Comment:
are we able to pipe this directly through? or do we need to read the whole thing into memory, hash it, and then reserialize it?

Reply:
Also if we did this chunk by chunk, we could keep a counter for the size, and if it gets above a certain threshold (ie the user lied about file size), then we throw an error